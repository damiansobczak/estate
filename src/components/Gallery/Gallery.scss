@import "../../assets/styles/utils/variables";
@import "../../assets/styles/utils/mixins";

.gallery {
  margin: 30px 0;
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  grid-template-rows: repeat(2, 250px);
  grid-gap: 5px;

  @include breakpoint-down(md) {
    grid-template-columns: repeat(2, 1fr);
  }

  @include breakpoint-down(sm) {
    grid-template-columns: repeat(1, 1fr);
    grid-auto-rows: 200px;
  }

  &__item {
    background-color: #f1f1f1;
    overflow: hidden;
    cursor: pointer;

    &:hover {
      img {
        transform: scale(1.1);
      }
    }

    @include breakpoint-up(md) {
      &:first-child {
        grid-column-start: 1;
        grid-column-end: 2;
        grid-row-start: 1;
        grid-row-end: 3;
      }

      &:last-child {
        grid-column-start: 3;
        grid-column-end: 4;
        grid-row-start: 1;
        grid-row-end: 3;
      }
    }

    img {
      width: 100%;
      height: 100%;
      object-fit: cover;
      transition: all 0.4s ease-in-out;
    }
  }
}
