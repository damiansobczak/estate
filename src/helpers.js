import Gallery1 from "./assets/image-1.jpg";
import Gallery2 from "./assets/image-2.jpg";
import Gallery3 from "./assets/image-3.jpg";
import Gallery4 from "./assets/image-4.jpg";

export const galleryAPI = [
  Gallery1, Gallery2, Gallery3, Gallery4
];

export const assetsAPI = [
  { id: 1, name: "Łazienka", desc: "Prysznic, pralka" },
  { id: 2, name: "Pokoje", desc: "2, TV, stół, sofa" },
  { id: 3, name: "Kuchnia", desc: "Kuchenka, czajnik" },
  { id: 4, name: "Taras", desc: "6m2, zastawa" },
  { id: 5, name: "Parking", desc: "Miejsce, szlaban" },
  { id: 6, name: "Metraż", desc: "26m2, taras" }
];

export const menuAPI = [
  { id: 1, name: "Strona główna", desc: "Biuro nieruchomości", url: "/" },
  {
    id: 2,
    name: "Wynajem",
    desc: "Strona z ofertami wynajmu",
    url: "/wynajem"
  },
  {
    id: 3,
    name: "Sprzedaż",
    desc: "Strona z ofertami sprzedaży",
    url: "/sprzedaz"
  }
];

export const offerJumboAPI = {
  subtitle: "Mieszkanie z garażem",
  title: "Przestronne mieszkanie, atrakcyjna lokalizacja",
  text:
    "Maecenas in turpis diam. Sed eget metus at urna mattis vehicula. Duis iaculis ligula a enim vestibulum lacinia. Sed eget metus at urna mattis vehicula. Duis iaculis ligula a enim. ",
  center: {
    x: 51.752549,
    y: 18.056496
  },
  address: "Ostrów Wielkopolski, Panorama ul. Jasna 32/7"
};
