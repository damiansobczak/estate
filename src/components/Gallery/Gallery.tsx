import React from "react";
import "./Gallery.scss";
import { GalleryProps } from "./Interfaces/GalleryProps";

class Gallery extends React.Component<GalleryProps> {
  render() {
    return (
      <section className="gallery">
        {this.props.images.map(image => (
          <div className="gallery__item" key={image.id}>
            <img src={image.src} alt={image.alt} />
          </div>
        ))}
      </section>
    );
  }
}

export default Gallery;
