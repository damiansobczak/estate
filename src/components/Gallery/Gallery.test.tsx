import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import Gallery from "./Gallery";
import { galleryAPI } from "../../helpers";

describe("<Gallery />", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Gallery images={galleryAPI} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders props correctly", () => {
    const wrapper = shallow(<Gallery images={galleryAPI} />);
    const images = wrapper.find(".gallery__item img");
    const imagesArr = images.map(el => el.props().src);
    expect(imagesArr.includes("galeria-1.png")).toBeTruthy();
  });
});
