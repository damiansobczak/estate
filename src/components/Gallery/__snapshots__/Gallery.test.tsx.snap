// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Gallery /> renders correctly 1`] = `
<section
  className="gallery"
>
  <div
    className="gallery__item"
  >
    <img
      alt="To jest obrazek galeri"
      src="galeria-1.png"
    />
  </div>
  <div
    className="gallery__item"
  >
    <img
      alt="To jest obrazek galeri"
      src="galeria-2.png"
    />
  </div>
  <div
    className="gallery__item"
  >
    <img
      alt="To jest obrazek galeri"
      src="galeria-3.png"
    />
  </div>
  <div
    className="gallery__item"
  >
    <img
      alt="To jest obrazek galeri"
      src="galeria-4.png"
    />
  </div>
</section>
`;
